CONTRIBUTING.md

```diff
@@ -26,17 +26,17 @@ Code contributions via pull requests are much appreciated. Before sending us a p
 
 1. You are working against the latest source on the *master* branch.
 2. You check existing open, and recently merged, pull requests to make sure someone else hasn't addressed the problem already.
-3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
+3. You open an issue to discuss any significant work before implementing it. Major new features, submodules, and model contributions need to fit into our overall design philosophy and our user's interests. We also need to evaluate if the contribution is worth the maintenance overhead.
 
 To send us a pull request, please:
 
-1. Fork the repository.
+1. Fork the repository and clone the source code to your local machine.
 2. Modify the source (see details below); please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
 3. Ensure local tests pass.
 4. Commit to your fork using clear commit messages.
 5. Send us a pull request, answering any default questions in the pull request interface.
 6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
-7. To spin up the platform tests, which test autogluon among macos and windows, comment on the PR with `/platform_tests`(You would need write permission to AutoGluon repo). It is recommended to run the platform tests only after you have passed the default CI.
+7. (Optional) To spin up the platform tests, which test autogluon among MacOS and Windows, comment on the PR with `/platform_tests`(You would need write permission to AutoGluon repo). It is recommended to run the platform tests only after you have passed the default CI and only for changes that are likely to cause platform specific issues.
```

Comment:
IIRC, one can only start platform tests from a branch of the `autogluon/autogluon` repo and not for branches of the forks. It makes sense for us (maintainers) to be aware of this, but probably not something for the users to worry about since they cannot run the tests.

Reply:
Good point, moved to a different section focusing on maintainer specific contribution instructions